### PR TITLE
Support for Maven reproducible builds

### LINKED ISSUE #103: Support for Maven reproducible builds
https://maven.apache.org/guides/mini/guide-reproducible-builds.html notes that [maven-release-plugin](https://maven.apache.org/plugins/maven-release-plugin/) version 3.0.0-M1 or later will automatically update the `project.build.outputTimestamp` property in `pom.xml` during the release in the same commit that updates version but that "if you have a custom release process tooling, you'll need to add the feature to your release tooling." This issue is blocking progress on jenkinsci/plugin-pom#919. One solution could be to set this to the timestamp of the most recent commit.

### PR DESCRIPTION
**Pull request by basil**
Add support for Maven reproducible builds when Maven Release Plugin is not in use, regardless of whether `project.build.outputTimestamp` is (as would be the case for a plugin that still used MRP for production releases but used Incrementals for PR releases) or is not (as would be the case for a plugin that fully implemented automated release) defined in `pom.xml`.

### Testing done

Verified that when passing `-Dset.changelist` Maven couldn't verify an incrementals build with `mvn clean verify artifact:compare` before this PR but could verify it after this PR.

Fixes #103

### CODE REVIEW

File: git-changelist-maven-extension/src/main/java/io/jenkins/tools/incrementals/git_changelist_maven_extension/Main.java [hunk 1]
```diff
@@ -100,6 +100,7 @@ public void afterSessionStart(MavenSession session) throws MavenExecutionExcepti
                     hash = head.abbreviate(ABBREV_LENGTH).name();
                     try (RevWalk walk = new RevWalk(repo)) {
                         RevCommit headC = walk.parseCommit(head);
+                        timestamp = DateTimeFormatter.ISO_INSTANT.format(Instant.ofEpochSecond(headC.getCommitTime()));
```
Comment (jglick) [inline git-changelist-maven-extension/src/main/java/io/jenkins/tools/incrementals/git_changelist_maven_extension/Main.java:103]: Should probably a string such as `2024-04-12T19:40:42Z` which seems to align with https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=74682318#Reproducible/VerifiableBuilds-OutputArchiveEntriesTimestamp though it is vague about allowed precision.

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
@basil I am ready to merge this and release 1.8 unless you have any last-minute requests?

**Comment by basil** [comment]
No last-minute requests on my side. Thank you!

**Comment by jglick** [comment]
https://github.com/jenkinsci/incrementals-tools/releases/tag/parent-1.8 ought to be available on Central shortly.

**Pull request merged**